src/Elastic.CommonSchema.Serilog/LogEventConverter.cs

```diff
@@ -85,55 +103,167 @@ public static Base ConvertToEcs(LogEvent logEvent, IEcsTextFormatterConfiguratio
 							.Replace("]", string.Empty)
 							.Split(','))
 						.Select(value => new { Key = value[0].Trim(), Value = value[1].Trim() }))
-						dict.Add(item.Key, item.Value);
+						dict.Add(ToSnakeCase(item.Key), item.Value);
 				}
 			}
 
 			foreach (var logEventPropertyValue in logEvent.Properties)
 			{
+				switch (logEventPropertyValue.Key) {
+					// already mapped as structured ECS event
+					case SpecialKeys.SourceContext:
+					case SpecialKeys.EnvironmentUserName:
+					case SpecialKeys.Host:
+					case SpecialKeys.ActionCategory:
+					case SpecialKeys.ActionName:
+					case SpecialKeys.ActionId:
+					case SpecialKeys.ActionKind:
+					case SpecialKeys.ActionSeverity:
+					case SpecialKeys.ApplicationId:
+					case SpecialKeys.ApplicationName:
+					case SpecialKeys.ApplicationType:
+					case SpecialKeys.ApplicationVersion:
+					case SpecialKeys.ProcessName:
+					case SpecialKeys.ProcessId:
+					case SpecialKeys.ThreadId:
+					case SpecialKeys.MachineName:
+						continue;
+				}
+
 				if (logEventPropertyValue.Value is SequenceValue values)
 				{
-					dict.Add(logEventPropertyValue.Key, values.Elements.Select(e => e.ToString()).ToArray());
+					dict.Add(ToSnakeCase(logEventPropertyValue.Key), values.Elements.Select(e => e.ToString()).ToArray());
 					continue;
 				}
-
-				dict.Add(logEventPropertyValue.Key, logEventPropertyValue.Value.ToString());
+				if (logEventPropertyValue.Value is ScalarValue sv)
+					dict.Add(ToSnakeCase(logEventPropertyValue.Key), sv.Value);
+				else
+					dict.Add(ToSnakeCase(logEventPropertyValue.Key), logEventPropertyValue.Value);
 			}
-
+			if (dict.Count == 0) return null;
 			return dict;
 		}
 
+		//TODO this should live in Log.MetaData as custom dictionary converter
+		private static string ToSnakeCase(string s)
```

Comment:
++ to this, but I also wonder if we ought to also store the original value before it was transformed?

`ValueX` -> `value_x`

But perhaps also store `original: "ValueX"` or something similar?

Reply:
This is now handled in the json converters in https://github.com/elastic/ecs-dotnet/pull/9

Since we only snake_case keys since all of  ECS is snake_case I don't think there is usecase to introduce the original key value.

Do you have a usecase in mind?

Reply:
The only reason was for if an external system (or user) wanted to correlate back to the original token name in the source code. I was thinking if we stored the original parameter name that would make it possible without having to UnSnakeCase (and potentially handle any mismatches)